### PR TITLE
Add type defs for "vosk"

### PR DESCRIPTION
**Pull request by TejasQ**
Just adding types for [vosk](https://www.npmjs.com/package/vosk).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] ~[Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.~
  - ⚠️ I'm just adding a `.d.ts` file with no logic. Is this still needed?
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/vosk/tslint.json
1. types/vosk/tsconfig.json
1. types/vosk/package.json

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2d4876558d4089dcfe2a4b8508a4544a06873fc5
</p>

File: types/vosk/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "compilerOptions": {
+        "target": "es2016",
+        "lib": ["es6"],
+        "module": "commonjs",
+        "types": [],
+        "noEmit": true,
+        "esModuleInterop": true,
```
Comment (gabritto) [inline types/vosk/tsconfig.json:8]: Please avoid using "esModuleInterop` in DT packages: https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

**Review by gabritto — CHANGES_REQUESTED** [review]
Please add tests: https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports.

Also, more importantly, I see that the original package already has type annotations in it, via JSDoc.
It would be better to get the original package to ship its own types, instead of copying them here on DefinitelyTyped, to make sure there's no duplication and the types don't get out of sync. I strongly encourage you to try doing that first, but if it's not possible, then it's ok to add the types here on DT.

File: types/vosk/tslint.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "extends": "@definitelytyped/dtslint/dt.json" }
```
Comment (jakebailey) [inline types/vosk/tslint.json:1]: You should remove this file; it is no longer required.

**Review by jakebailey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**